src/modules/ekf2/EKF/aid_sources/magnetometer/mag_control.cpp

```diff
@@ -69,12 +69,22 @@ void Ekf::controlMagFusion(const imuSample &imu_sample)
 
 			_state.mag_B.zero();
 			resetMagBiasCov();
+			_state.mag_I.zero();
```

Comment:
Should we push most of this into stopMagFusion() instead?

Reply:
This should only be done if there is a change in mag calibration, not every time we stop the mag. If the mag fusion gets stopped for other reasons, we actually want to keep the current state estimates I think.

Reply:
After more testing, I found that this `mag_I` reset is not necessary, also the covariance reset is already in the `stopMagFusion` code so I just removed both.